app/Http/Controllers/TasksController.php

```diff
@@ -57,9 +57,16 @@ public function show($id)
      * @param  int  $id
      * @return \Illuminate\Http\Response
      */
-    public function edit($id)
+    public function edit(Request $request)
     {
-        //
+        $task = Task::find($request->segment(3));
```

Comment:
a dùng findorfail được không và nó khác nhau như nào

Reply:
> a dùng findorfail được không và nó khác nhau như nào

nếu query thành công thì find() và findOrFail() trả về cùng 1 kết quả, nếu không thành công thành find() trả về giá trị null còn finOrFail() trả về trang 404. Ở đây em muốn nếu không query thành công sẽ return về trang index và thông bao cho người dùng nên em sử dụng find() mà không sử dụng findOrFail().

Reply:
> a dùng findorfail được không và nó khác nhau như nào

nếu query thành công thì cả 2 đều trả về cùng 1 kết quả, nhưng nếu không thành công thì find() trả về null, findorfail() throw ra 1 ModelNotFoundException, mình có thể catch Exception đó để xử lí.
Nhưng vấn đề ở đây là thông báo lỗi của Exception này có dạng "No query results for model [App\Models\Task] 30". Thông báo này không phù hợp để show ra cho người dùng thấy được nên ta phải tự tạo 1 thông báo thân thiện hơn cho người dùng, thành ra mình cũng không tận dụng được Exception đó. Nên em thấy trong trường hợp này find hay findorfail đều được .